pkg/grpc/requestinfo/requestinfo.go

```diff
@@ -269,7 +275,12 @@ func (h *Handler) UpdateContextForGRPC(ctx context.Context) (context.Context, er
 		ri.VerifiedChains = ExtractCertInfoChains(tlsState.VerifiedChains)
 	}
 
-	ri.Metadata, _ = metadata.FromIncomingContext(ctx)
+	md, _ := metadata.FromIncomingContext(ctx)
```

Comment:
Hmm, why go through the hoops of creating the custom type and all if we can just create a `metadata.MD` that removes the `runtime.MetadataPrefix` always from each key? This way we don't have to implement the custom type and all functions, but streamline things and keep the interface as-is.

Also, can you remind me for which headers the prefix will be set? It won't be set for all headers, so curious if we figured out which ones.

Reply:
For this generic package I thought I'd preserve the original map with its keys untouched, so they would still be available for whatever use via type cast, if needed.
The `grpcgateway-` prefix is set by the gateway for the "permanent" (i.e. standard IANA) header keys. HTTP headers that start with `Grpc-Metadata-` are mapped to gRPC metadata after removing prefix `Grpc-Metadata-` (see [DefaultHeaderMatcher](https://pkg.go.dev/github.com/grpc-ecosystem/grpc-gateway/v2/runtime#DefaultHeaderMatcher)).
I'm going to reuse the `DefaultHeaderMatcher` to query metadata passed via the gateway.

Reply:
In the next PR #10222 I'm introducing a `HeaderSetter` interface, so that we could set headers or metadata in a [generic way](https://github.com/stackrox/stackrox/pull/10222/files#diff-c571d80a0e339a61310a8c3e85b5217acdc90708e6d8c8b325be55500919e4c7R114).